### ISSUE TITLE
Running on 4 V100s, but epoch stays at 0

### ISSUE THREAD

**Issue by slerman12** [wilson1yan/VideoGPT#23]
I'm running the model on 4 V100 GPUs using SLURM and the following sbatch script:

```
#!/bin/bash
#SBATCH -p gpu
#SBATCH --gres=gpu:4
#SBATCH -p reserved --reservation=slerman-20210821 -t 2-00:00:00 
#SBATCH -t 5-00:00:00 -o ./vgpt.log -J vgpt
#SBATCH --mem=50gb 
#SBATCH -C V100
source /scratch/slerman/miniconda/bin/activate vid
python3 train_videogpt.py --max_steps 200000 --vqvae ucf101_stride4x4x4 --data_path ./datasets/ucf101/ --gpus 4
```

However, after a full day, the logs still show the model stuck at epoch 0.

Do you know what's going wrong?

Thank you.

**Comment by slerman12** [comment]
I'll try adding those val_check and limit_val commands. It's so strange, a single Epoch actually took >50 hrs:

Epoch 0: 100%|██████████| 71360/71360 [50:55:33<00:00,  2.57s/it, loss=3.64, v_num=6635713, val/loss=3.690]

**Comment by slerman12** [comment]
This is on 4 V100s

**Comment by wilson1yan** [comment]
Hmm the 2.56s/it actually doesn't seem too off. You can get a good amount of speed-up (2x or more) if you train with sparse attention and mixed precision: `attn_type 'sparse' --amp_level O1 --precision 16`. Though I've found that sometimes the mixed precision is unstable depending on what kind of model is being trained (i.e. usually works with unconditional, but seems to be more unstable when training class-conditional).

**Comment by ebuss81** [comment]
hello all, i have a similar problem. i use the bair dataset and the training process ends after batchsize + 1 iterations. Any idea what the problem is?

Example: Batchsize = 16 

python train_vqvae.py --data_path bair.hdf5 --gpus 2 --batch_size 2 --sequence_length 16 --accumulate 8 --val_check_interval 5000 --limit_val_batches 500

Epoch 0:   0%|                                                                                                                                                                                                                                                                          | 17/22016 [00:06<2:21:25,  2.59it/s, loss=1.74, v_num=60]

Example: Batchsize = 32

 python train_vqvae.py --data_path bair.hdf5 --gpus 2 --batch_size 2 --sequence_length 16 --accumulate 16 --val_check_interval 5000 --limit_val_batches 500

Epoch 0:   0%|▍                                                                                                                                                                                                                                                                                                                                 | 33/22016 [00:10<1:54:09,  3.21it/s, loss=1.98, v_num=63]


Thanks in advance!
